L1Trigger/RPCTechnicalTrigger/src/RBCEmulator.cc

```diff
@@ -15,77 +15,46 @@
 //=============================================================================
 // Standard constructor, initializes variables
 //=============================================================================
-RBCEmulator::RBCEmulator( ) {
-  
-  m_signal  = nullptr;
-  m_logtype = std::string("TestLogic");
-  m_rbcinfo = new RBCId();
-  m_input   = new RBCInput();
-    
-  m_layersignal[0] = new std::bitset<6>();
-  m_layersignal[1] = new std::bitset<6>();
-  m_layersignalVec.push_back( m_layersignal[0] );
-  m_layersignalVec.push_back( m_layersignal[1] );
```

Comment:
Here,` m_layersignalVec` is actually taking ownership of the memory. `m_layersignal` may have its pointer redirected to another value where it does not own that memory. In my version of the code, I tried to make that behavior more obvious by showing the assignment.